File: src/dataset/go/resistor_color_duo_Exercism_go_thinkverse/gemini/GeminiFlash_resistor_color_duo_v3.go
package resistorcolorduo

var colors = []string{
	"black",
	"brown",
	"red",
	"orange",
	"yellow",
	"green",
	"blue",
	"violet",
	"grey",
	"white",
}

// Value should return the resistance value of a resistor with a given colors.
func Value(inputColors []string) int {
	if len(inputColors) < 2 {
		return 0
	}

	color1, color2 := inputColors[0], inputColors[1]
	digit1, digit2 := 0, 0

	for i, color := range colors {
		if color == color1 {
			digit1 = i
		}
		if color == color2 {
			digit2 = i
		}
	}

	return (digit1 * 10) + digit2
}

File: src/dataset/javascript/resistor-color-duo_exercism-javascript-oguzsh/gemini/GeminiFlash_resistor-color-duo_v3.js
export const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white"
];

const COLOR_MAP = new Map(COLORS.map((color, index) => [color, index]));

export const decodedValue = colors => {
  const digit1 = COLOR_MAP.get(colors[0]);
  const digit2 = COLOR_MAP.get(colors[1]);

  return digit1 * 10 + digit2;
};